extensions/azurePublishNew/src/recoilModel/selectors/tenants.ts

```diff
@@ -0,0 +1,15 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { usePublishApi } from '@bfc/extension-client';
+import { selector } from 'recoil';
+
+export const tenantIdSelector = selector({
+  key: 'tenantIdSelector',
+  get: () => {
+    const { userShouldProvideTokens, getTenantIdFromCache } = usePublishApi();
```

Comment:
this is against react hook rules, does this somehow work?!
A hook can only be used within the context of a functional component or another custom hook.

@a-b-r-o-w-n can you chime in?

Reply:
I see what you are saying, usePublishApi() is not actually a hook, it is just exports a bunch of functions, please correct me if I am wrong.